### PR TITLE
dont detach on autoattach

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Test Steps

Run edited integration test.

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: docs/source/tutorials/create_a_fips_updates_pro_cloud_image.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+You will need to re-enable fips-updates on each instance that is launched from the custom image.
+
+This won't require a reboot and is only necessary to ensure the instance gets updates to fips packages when they become available.
+
+```bash
+sudo ua enable fips-updates --assume-yes
+```
+
+You can easily script this using [cloud-init](https://cloudinit.readthedocs.io/en/latest/topics/modules.html#runcmd)
```
Comment (blackboxsw) [inline docs/source/tutorials/create_a_fips_updates_pro_cloud_image.md:52]: Might as well just provide that for folks:
```suggestion
You can easily script with the user-data at launch time using [cloud-init](https://cloudinit.readthedocs.io/en/latest/topics/modules.html#runcmd)
```
#cloud-config
runcmd:
 - 'ua status --wait'
 - 'ua enable fips-updates --assume-yes'
power_state:
  mode: reboot
```

```

File: uaclient/cli.py [hunk 2]
```diff
@@ -1340,17 +1343,6 @@ def action_auto_attach(args, *, cfg):
...
-    if cfg.is_attached:
-        prev_iid = cfg.read_cache("instance-id")
-        if str(current_iid) == str(prev_iid):
-            raise exceptions.AlreadyAttachedOnPROError(str(current_iid))
-        print("Re-attaching Ubuntu Pro subscription on new instance")
-        if _detach(cfg, assume_yes=True) != 0:
-            raise exceptions.UserFacingError(
-                messages.DETACH_AUTOMATION_FAILURE
-            )
-
```
Comment (blackboxsw) [inline uaclient/cli.py:1353]: I think this is reasonable given the change in the contract server hanlding which disambiguates multiple attached machines under the same contract. +1

**Review by blackboxsw — APPROVED** [review]
This behavior, documentation and integration testss look good to me. Note as well that cloud-init is adding a `cloud-init clean --machine-id` parameter to help folks with image cloning in general adhere to best practices to ensure new boots of cloned images create unique /etc/machine-id on next boot. I realize this isn't quite what you are getting at with your doc example here. But wanted to FYI on that option that landed in tip of main for cloud-init.

File: docs/source/tutorials/create_a_fips_updates_pro_cloud_image.md (hunk 1 above)
Comment (orndorffgrant) [reply docs/source/tutorials/create_a_fips_updates_pro_cloud_image.md:52]: The reboot is specifically not required, but point taken! I'll add the runcmd config to the doc.

**Comment by orndorffgrant** [comment]
Thanks @blackboxsw for the review! Do you think we should suggest cleaning the machine-id in our docs? If that is a generally recommended practice, we should probably include it.

**Review by renanrodrigo — APPROVED** [review]
lgtm
:shipit:

**Comment by blackboxsw** [comment]
> 

I'd say we want to suggest `rm /etc/machine-id` if creating a cloned image to ensure new machine-ids are created and cloned images don't have to rely on the contract backend disambiguation. What you'd lose by suggesting that step is a simple way in contract backend to track how many clones of a PRO image are out there. So that might be a thing the talk about with Product as to whether that is a "feature" that is wanted.

**Comment by orndorffgrant** [comment]
That makes sense @blackboxsw! I added the `rm /etc/machine-id` step to our docs

merging now

**Pull request merged**